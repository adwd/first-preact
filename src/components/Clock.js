import { h, render, Component } from 'preact'
import styles from './Clock.scss'

export default class Clock extends Component {
  render({ date }) {
    return (
      <div>
        <p>time is</p>
        <span class={styles.clock}>{date.toLocaleTimeString()}</span>
      </div>
    )
  }
}
