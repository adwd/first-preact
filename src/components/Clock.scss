.clock {
  padding: 8px;
  background-color: lightblue;
}
