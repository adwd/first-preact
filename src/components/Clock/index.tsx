import { h, render, Component } from 'preact'
import * as styles from './style.scss'

export default class Clock extends Component<{ date: Date }, {}> {
  render({ date }: { date: Date }) {
    return (
      <div>
        <span class={styles.clock}>{date.toLocaleTimeString()}</span>
      </div>
    )
  }
}
