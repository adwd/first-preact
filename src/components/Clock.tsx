import { h, render, Component } from 'preact'
import * as styles from './Clock.scss'

export default class Clock extends Component<{ date: Date }, {}> {
  render({ date }: { date: Date }) {
    return (
      <div>
        <p>time is</p>
        <span class={styles.clock}>{date.toLocaleTimeString()}</span>
      </div>
    )
  }
}
