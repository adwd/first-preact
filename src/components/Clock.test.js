import { h, Component } from 'preact'
import Clock from './Clock'
import { shallow } from 'preact-render-spy'

describe('Clock', () => {
  test('renders clock', () => {
    const date = new Date(2018, 1, 1, 11, 12, 13, 14)
    const context = shallow(<Clock date={date} />)
    expect(context.find('p').text()).toBe('time is')
    expect(context.find('span').text()).toBe('11:12:13')
  })
})
