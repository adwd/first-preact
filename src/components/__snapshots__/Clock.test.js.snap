// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<div>
  <p>time is</p>
  <span>11:12:13</span>
</div>

`;
